src/main/java/gregtech/integration/jei/multiblock/MultiblockInfoPage.java

```diff
@@ -1,14 +1,26 @@
 package gregtech.integration.jei.multiblock;
 
+import gregtech.api.GTValues;
 import gregtech.api.metatileentity.multiblock.MultiblockControllerBase;
+import gregtech.api.util.ItemStackHashStrategy;
+import gregtech.common.metatileentities.MetaTileEntities;
+import it.unimi.dsi.fastutil.Hash;
+import it.unimi.dsi.fastutil.objects.Object2ObjectOpenCustomHashMap;
 import net.minecraft.client.resources.I18n;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.Tuple;
+import net.minecraft.util.text.TextFormatting;
 
-import java.util.List;
+import java.util.*;
 import java.util.stream.Collectors;
 import java.util.stream.Stream;
 
 public abstract class MultiblockInfoPage {
 
+    private final Hash.Strategy<ItemStack> strategy = ItemStackHashStrategy.comparingAllButCount();
+
+    private Map<ItemStack, List<Tuple<String, TextFormatting>>> abilityTooltips = new Object2ObjectOpenCustomHashMap<>(strategy);
```

Comment:
This is no longer an ability keyed tooltip.
You need to update the name and some javadocs.